modulemd/modulemd-util.c

```diff
@@ -145,3 +145,100 @@ modulemd_variant_unref (void *ptr)
 {
   g_variant_unref ((GVariant *)ptr);
 }
+
+gboolean
+modulemd_validate_nevra (const gchar *nevra)
+{
+  g_autofree gchar *tmp = g_strdup (nevra);
+  gsize len = strlen (nevra);
+  gchar *i;
+  gchar *endptr;
+
+  /* Since the "name" portion of a NEVRA can have an infinite number of
+   * hyphens, we need to parse from the end backwards.
+   */
+  i = &tmp[len - 1];
+
+  /* Everything after the last '.' must be the architecture */
```

Comment:
strrchr? or there probably some glib version

Reply:
I don't really see how that's more advantageous.